src/screens/updatePlanItem/PlanItemTimer.tsx

```diff
@@ -2,67 +2,62 @@ import React from 'react';
 import {
   StyleSheet,
   Text,
-  TimePickerAndroid,
   TouchableOpacity,
 } from 'react-native';
 
-import { Icon, StyledText } from 'components';
+import { Icon } from 'components';
 import { PlanItem } from 'models';
+import { palette } from 'styles';
+import { TimePicker } from './TimePicker';
 
 interface Props {
   planItem: PlanItem;
 }
 
 interface State {
-  itemTime: number;
-  itemTimeText: string;
+  visible: boolean;
 }
 
 export class PlanItemTimer extends React.PureComponent<Props, State> {
+  minutesPicker: any;
+  secondsPicker: any;
   constructor(props: Props) {
     super(props);
     this.state = {
-      itemTime: this.props.planItem.time,
-      itemTimeText: (this.props.planItem.time!!) 
-        ? (this.props.planItem.time / 60).toFixed() + ':' + this.props.planItem.time % 60
-        : ''
+      visible: false,
     };
   }
 
-  pickTime = async () => {
-    try {
-      // @ts-ignore
-      const { action, hour, minute } = await TimePickerAndroid.open({
-        hour: 0,
-        minute: 5,
-        is24Hour: true,
-        mode: 'spinner',
-      });
-      if (action !== TimePickerAndroid.dismissedAction) {
-        this.setState({
-          itemTimeText: hour + ':' + minute,
-        });
-        // Now it stores time in minutes but picker will allow minutes and seconds
-        // firebase will store seconds
-        if (action === TimePickerAndroid.timeSetAction) {
-          this.props.planItem.update({
-            time: ((hour * 60) + minute),
-          });
-        }
-      }
-    } catch ({ code, message }) {
-      // tslint:disable-next-line:no-console
-      console.warn(message);
-    }
-  };
+  itemTimeText = () => {
+    return (
+      Math.floor(this.props.planItem.time / 60) + ':' 
+      + ((this.props.planItem.time % 60 < 10) ? '0' : '' )
+      + this.props.planItem.time % 60
+    );
+  }
+  
+  updatePlanItemTime = (minutes: number, seconds: number) => {
+    this.props.planItem.update({
+      time: minutes*60 + seconds
+    });
+  }
 
-  render() {
-    const { itemTimeText } = this.state;
+  wasTimerPressed = () => this.state.visible;
 
+  render() {
     return (
-      <TouchableOpacity onPress={this.pickTime}>
+      <TouchableOpacity onPress={() => this.setState({visible: true})}>
```

Comment:
Please do not include lamdas code inside render method. The code is violating `jsx-no-lambda` rule. To fix it you can create the state change handler above the render method.

```
setVisible = () => this.setState({ visible: true})
```